apps/artemis-web/components/hijack-info/hijack-info.tsx

```diff
@@ -182,59 +184,77 @@ class HijackInfoComponent extends Component<any, any> {
-                      >
-                        {!isResolved(this.props.hijackDataState) &&
-                          !isIgnored(this.props.hijackDataState) && (
-                            <>
-                              <option value="hijack_action_resolve">
-                                Mark as Resolved
+                      {this.user && this.user.role === 'admin' ?
+                        (
+                          <select
+                            ref={this.selectRef}
+                            style={{
+                              width: '200px',
+                              display: 'inline-block',
+                              marginRight: '15px',
+                            }}
+                            className="form-control form-control-sm-auto"
+                            id="action_selection"
+                          >
+                            {!isResolved(this.props.hijackDataState) &&
+                              !isIgnored(this.props.hijackDataState) && (
+                                <>
+                                  <option value="hijack_action_resolve">
+                                    Mark as Resolved
+                                  </option>
+                                  <option value="hijack_action_ignore">
+                                    Mark as Ignored
+                                  </option>
+                                </>
+                              )}
+                            {!isUnderMitigation(this.props.hijackDataState) ? (
+                              <option value="hijack_action_mitigate">
+                                Mitigate Hijack
                               </option>
-                              <option value="hijack_action_ignore">
-                                Mark as Ignored
+                            ) : (
+                              <option value="hijack_action_unmitigate">
+                                Un-mitigate Hijack
                               </option>
-                            </>
-                          )}
-                        {!isUnderMitigation(this.props.hijackDataState) ? (
-                          <option value="hijack_action_mitigate">
-                            Mitigate Hijack
-                          </option>
-                        ) : (
-                          <option value="hijack_action_unmitigate">
-                            Un-mitigate Hijack
-                          </option>
-                        )}
-                        {!isSeen(this.props.hijackDataState) ? (
-                          <option value="hijack_action_acknowledge">
-                            Mark as Acknowledged
-                          </option>
+                            )}
+                            {!isSeen(this.props.hijackDataState) ? (
+                              <option value="hijack_action_acknowledge">
+                                Mark as Acknowledged
```

Comment:
The Ack and Not Ack actions should also be allowed to a non-admin user (it is OK with our ARTEMIS audience).
So a non-admin use should be able to ack/not ack, plus export hijack info. All the other actions are done by an admin, as you have implemented the check now.

Reply:
fixed :) To do that I changed the rmq_action endpoint permissions and added rmq_admin_action for the full set of actions